app/src/main/java/samyups/example/trainingss/ui/MainRecyclerAdapter.kt

```diff
@@ -0,0 +1,34 @@
+package samyups.example.trainingss.ui
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import samyups.example.trainingss.R
+import samyups.example.trainingss.data.Dogs
+
+class MainRecyclerAdapter : RecyclerView.Adapter<MainViewHolder>() {
+
+    private var dogNamesList = emptyList<Dogs>()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MainViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.recycler_view_item, parent, false)
+        return MainViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: MainViewHolder, position: Int) {
+        val dogBreed = dogNamesList[position].breed
+        val dogWeight = dogNamesList[position].weight
+        val dogHeight = dogNamesList[position].height
+        val dogMaxAge = dogNamesList[position].maxAge
+        val dogData = Dogs(dogBreed, dogWeight, dogHeight, dogMaxAge)
```

Comment:
This seems redundant. You access `dogNamesList[position]` which gives you an instance of the `Dogs` object. Then you extract each field from the object (`breed`, `weight`, etc), then you rebuild the exact same `Dogs` object using those fields and then pass it to the viewholder. Why not just pass the original `Dogs` object to the viewholder?

```
val dog = dogNamesList.position]
holder.bind(dog)
```